internal/core/domain/deposit_repository.go

```diff
@@ -0,0 +1,17 @@
+package domain
+
+import "context"
+
+type Page struct {
+	Number int
+	Size   int
+}
```

Comment:
Would be better to be in a dedicated page_model.go file. Also, we should add a method Validate to this type that make sure that the page is valid: Number and Size MUST be both greater than 0 (badger repo would panic otherwise in such cases). This method should be ALWAYS called by any repo that uses Page to prevent unexpected errors/panics.

Reply:
Done